### PR TITLE
Added specs and updates to wait.html test page to support when_enabled.

### PR DESCRIPTION
**Pull request by Rodney-QAGeek**
Tests to support newly proposed method when_enabled added here https://github.com/watir/watir-webdriver/pull/363

### CODE REVIEW

File: wait_spec.rb [hunk 1]
```diff
@@ -140,6 +140,64 @@ def present?
...
+
+      not_compliant_on :watir_classic do
+        it "times out when not given a block" do
+          expect { browser.button(id: 'btn').when_enabled(1).click }.to raise_error(Watir::Wait::TimeoutError,
+            /^timed out after 1 seconds, waiting for (\{:id=>"btn", :tag_name=>"button"\}|\{:tag_name=>"button", :id=>"btn"\}) to become enabled$/
+          )
+        end
+      end
+
+      deviates_on :watir_classic do
```
Comment (p0deje) [inline wait_spec.rb]: I believe this is `not_compliant`, not `deviates`

File: wait_spec.rb [hunk 2]
```diff
@@ -140,6 +140,64 @@ def present?
...
+          expect { browser.button(id: 'btn').when_enabled(1).click }.to raise_error(Watir::Wait::TimeoutError,
+            /timed out after 1 seconds, waiting for {:id=>"btn", :tag_name=>"button"} to become enabled$/
+          )
+        end
+      end
+
+      it "responds to Element methods" do
+        decorator = browser.button.when_enabled
+
+        decorator.should respond_to(:exist?)
```
Comment (p0deje) [inline wait_spec.rb]: Use expect syntax:

```ruby
expect(decorator).to respond_to(:exist?)
```

File: wait_spec.rb [hunk 3]
```diff
@@ -140,6 +140,64 @@ def present?
...
+        btn = browser.button(id: 'btn')
+        btn.when_enabled(2).click
+        expect(btn.disabled?).to be true
+      end
+
+      it "times out when given a block" do
+        expect { browser.button(id: 'btn').when_enabled(1) {}}.to raise_error(Watir::Wait::TimeoutError)
+      end
+
+      not_compliant_on :watir_classic do
```
Comment (p0deje) [inline wait_spec.rb]: Why is it not compliant on watir-classic?

File: wait_spec.rb [hunk 4]
```diff
@@ -140,6 +140,64 @@ def present?
...
     end
 
+    describe "#when_enabled" do
+      it "yields when the element becomes enabled" do
+        called = false
+
+        browser.a(id: 'enable_btn').click
+        browser.button(id: 'btn').when_enabled(2) { called = true }
+
+       expect(called).to be true
```
Comment (p0deje) [inline wait_spec.rb]: Indent please

**Comment by p0deje** [comment]
It would be great if you could rebase the commits here to have a single commit with the changes and without all theses merge commits.

**Comment by Rodney-QAGeek** [comment]
Thanks I'll make these changes now!

**Pull request closed without merging**